seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/java/org/apache/seatunnel/e2e/transform/TestDynamicCompileIT.java

```diff
@@ -27,47 +27,65 @@
 
 public class TestDynamicCompileIT extends TestSuiteBase {
 
+    private final String basePath = "/dynamic_compile/conf/";
+
     @TestTemplate
     public void testDynamicSingleCompileGroovy(TestContainer container)
             throws IOException, InterruptedException {
         Container.ExecResult execResult =
-                container.executeJob(
-                        "/dynamic_compile/single_dynamic_groovy_compile_transform.conf");
+                container.executeJob(basePath + "single_dynamic_groovy_compile_transform.conf");
         Assertions.assertEquals(0, execResult.getExitCode());
     }
 
     @TestTemplate
     public void testDynamicSingleCompileJava(TestContainer container)
             throws IOException, InterruptedException {
         Container.ExecResult execResult =
-                container.executeJob("/dynamic_compile/single_dynamic_java_compile_transform.conf");
+                container.executeJob(basePath + "single_dynamic_java_compile_transform.conf");
         Assertions.assertEquals(0, execResult.getExitCode());
     }
 
     @TestTemplate
     public void testDynamicMultipleCompileGroovy(TestContainer container)
             throws IOException, InterruptedException {
         Container.ExecResult execResult =
-                container.executeJob(
-                        "/dynamic_compile/multiple_dynamic_groovy_compile_transform.conf");
+                container.executeJob(basePath + "multiple_dynamic_groovy_compile_transform.conf");
         Assertions.assertEquals(0, execResult.getExitCode());
     }
 
     @TestTemplate
     public void testDynamicMultipleCompileJava(TestContainer container)
             throws IOException, InterruptedException {
         Container.ExecResult execResult =
-                container.executeJob(
-                        "/dynamic_compile/multiple_dynamic_java_compile_transform.conf");
+                container.executeJob(basePath + "multiple_dynamic_java_compile_transform.conf");
         Assertions.assertEquals(0, execResult.getExitCode());
     }
 
     @TestTemplate
     public void testDynamicMixedCompileJavaAndGroovy(TestContainer container)
             throws IOException, InterruptedException {
         Container.ExecResult execResult =
-                container.executeJob(
-                        "/dynamic_compile/mixed_dynamic_groovy_java_compile_transform.conf");
+                container.executeJob(basePath + "mixed_dynamic_groovy_java_compile_transform.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+    }
+
+    @TestTemplate
+    public void testDynamicSinglePathGroovy(TestContainer container)
+            throws IOException, InterruptedException {
+        container.copyFileMyselfContainers(
+                "/dynamic_compile/source_file/GroovyFile", "/tmp/GroovyFile");
```

Comment:
Do we need to save the same code file on all nodes? Maybe we can read the file into a String value in the constructor, so that we only need to keep one code file on the client.
https://github.com/apache/seatunnel/pull/7264/files#diff-232afd1e749a56f59c831ec14b761e25bb712e42a52001e8c47917dcc975edb7R48-R60

Reply:
Thank you for your suggestion @Hisoka-X  Done PTAL